bolt-modules/boltlib/lib/puppet/datatypes/target.rb

```diff
@@ -3,54 +3,33 @@
 Puppet::DataTypes.create_type('Target') do
   begin
     inventory = Puppet.lookup(:bolt_inventory)
-
-    inventory_version = inventory.version
-    if inventory_version != 1
-      target_implementation_class = inventory.target_implementation_class
-    end
+    target_implementation_class = inventory.target_implementation_class
```

Comment:
I think we can just get rid of this logic and hardcode the implementation class.

Reply:
nm this is used to choose the special apply class.

Reply:
We need to keep it since 2.0 has `Bolt::ApplyInventory` and `Bolt::ApplyTarget`.